src/Markup/Avalonia.Markup/Data/Binding.cs

```diff
@@ -59,20 +59,21 @@ public Binding(string path, BindingMode mode = BindingMode.Default)
         /// <summary>
         /// Gets or sets a function used to resolve types from names in the binding path.
         /// </summary>
-        public Func<string, string, Type> TypeResolver { get; set; }
+        public Func<string, string, Type>? TypeResolver { get; set; }
 
-        protected override ExpressionObserver CreateExpressionObserver(IAvaloniaObject target, AvaloniaProperty targetProperty, object anchor, bool enableDataValidation)
+        protected override ExpressionObserver CreateExpressionObserver(IAvaloniaObject target, AvaloniaProperty targetProperty, object? anchor, bool enableDataValidation)
         {
-            Contract.Requires<ArgumentNullException>(target != null);
+            Contract.Requires<ArgumentNullException>(target is not null);
```

Comment:
Note that when I add nullable reference types, I remove `Contract.Requires` and replace it with `arg = arg ?? throw new ArgumentNullException(nameof(arg))` because:

- It's nearly as succinct
- It plays nicer with NRTs
- It actually includes the name of the arg in the error message

Reply:
I tried to keep changes minimal, but I can change that (also because `is not` is a C# 9 feature and it breaks the build).